hash.c

```diff
@@ -1587,6 +1598,12 @@ hash_copy(VALUE ret, VALUE hash)
     return ret;
 }
 
+static VALUE
+hash_copy_with_compare_by_id(VALUE ret, VALUE hash)
```

Comment:
FYI, all calls have `ret = rb_hash_new()`, so maybe `hash_dup_with_compare_by_id(hash)` would be a better signature.

Reply:
Agreed, I'll push a commit for that shortly.